### PR TITLE
mozlz4a: enable build on Darwin

### PR DESCRIPTION
**Pull request by pshirshov**
See https://github.com/NixOS/nixpkgs/issues/196018

### CODE REVIEW

File: pkgs/tools/compression/mozlz4a/default.nix
Comment (ncfavier) [inline pkgs/tools/compression/mozlz4a/default.nix:30]:
```suggestion
    platforms = lib.platforms.unix;
```

**Review by Atemu — APPROVED** [review]
Successfully decompresses my profile's search.json.mozlz4a.

Could you add yourself as a maintainer of this package for the Darwin part?

Result of `nixpkgs-review pr 196021` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mozlz4a</li>
  </ul>
</details>

**Review by kira-bruneau — APPROVED** [review]
Thanks @pshirshov! Since I plan on maintaining the [search portion of the firefox home-manager module](https://github.com/nix-community/home-manager/pull/2984), you can add me as a darwin maintainer if you don't want to maintain it yourself.

(I have a x86_64-darwin mac to test on)

**Comment by Atemu** [comment]
@kira-bruneau please check new commit messages before merging. They do not fit the contributor guidelines.

**Comment by kira-bruneau** [comment]
Oh, sorry, I missed that! :sweat:

**Comment by pshirshov** [comment]
I think that squash on merge should be enabled for the repo.

**Comment by kira-bruneau** [comment]
@pshirshov I don't think we'd want to enable that, because in certain cases it makes sense to have multiple commits in the PR, and we don't want to loose that context.

The normal workflow in nixpkgs is to cleanup your history locally where it makes sense (eg. amend commits instead of adding "fixup" commits) and then force push your changes.

**Comment by ncfavier** [comment]
Here's the comment where this was first explained to me: https://github.com/NixOS/nixpkgs/pull/104836#issuecomment-736123877

**Comment by pshirshov** [comment]
Yeah, but it's really convenient not to have local copy when I do some small changes in a nix formula. Also it's possible to allow squashes together with other policies.

**Pull request merged**
